### ISSUE TITLE
Validate nested hash with integer keys

### ISSUE THREAD

**Issue by adam12** [aradunovic/fend#1]
First off, very interesting project!

I've been evaluating a few different options and I notice many of them fall down in the following scenario, which is fairly common when receiving POST data from a form which accepts params for multiple objects (`accepts_nested_attributes_for` Rails kind).

I've looked and don't see an obvious way to achieve it with Fend. Any ideas?

```ruby
require "pp"
require "fend"

data = {
  "album" => {
    "title" => "The Album",

    "tracks_attributes" => {
      "0" => { "title" => "The First Track", "runtime" => "2:35" },
      "1" => { "title" => "The Second Track", "runtime" => "2:20" }
    }
  }
}

class AlbumValidation < Fend
  plugin :validation_helpers

  validate do |i|
    i.params("title") { |title| title.validate_presence }
    i.params("tracks_attributes") do |tracks_attributes|
      # How to validate numeric keys with hash values
    end
  end
end

AlbumValidation.new.call(data.fetch("album"))
```

**Comment by aradunovic** [comment]
Hi Adam,

Thank you very much for trying out Fend!!!

I just pushed a [minor improvement](https://github.com/aradunovic/fend/commit/243bb33af92c2aa4eec1967379b41de91447fc7e) to master, which should cover the scenario you described.
As of now, you can use `#each` method with `hash: true` option in order to iterate over nested hash params while ignoring the keys/names:

```ruby
validate do |i|
  i.params("title") { |title| title.validate_presence }

  i.params("tracks_attributes") do |tracks_attributes|
    
    tracks_attributes.each(hash: true) do |track|
      track.params("title", "runtime") do |title, runtime|
        # validate title & runtime
      end
    end

  end
end
```

Error messages will be nested under unmodified keys - in your case `"0", "1"`.
```ruby
result.messages #=> { tracks_attributes: { "0" => { title: ["error"], runtime: ["error"] } } }
```
This also covers the cases where instead of 
`"1" => { title: "Foo" }` 
there is something like
`"track_2" => { title: "Foo" }`.

This feature will be available in v0.3.0. Until then, feel free to install from master branch.
If you have any other questions or even suggestions, feel free to contact me.

Also, please let me know if this actually solves your problem 😄 

Cheers!

**Comment by adam12** [comment]
This worked a treat. Much thanks :)
